### PR TITLE
Allow double digits in major version

### PR DESCRIPTION
**Pull request by imurchie**
The first beta of Xcode 10 is out, and our regex doesn't allow for it.

### CODE REVIEW

**Review by mykola-mokhnach — APPROVED** [review]

**Comment by mykola-mokhnach** [comment]
Can we use semver instead?

**Comment by imurchie** [comment]
Not right here, no, since we aren't parsing `10.0`, but pulling `10.0` out of the string `Xcode 10.0`.

**Comment by jlipps** [comment]
Probably the semver library has a string-to-semver constructor, though?

**Comment by imurchie** [comment]
coerce might work. I'll give it a shot.

**Comment by imurchie** [comment]
Alrighty. Semver all the things.

**Review by jlipps — CHANGES_REQUESTED** [review]
nice!

File: lib/xcode.js [hunk 1]
```diff
@@ -136,20 +128,20 @@ const getVersionMemoized = _.memoize(
...
-  //     [ '7.0.1', '7', '0', '.1', '1', index: 0, input: '7.0.1' ]
+
   return {
-    versionString: version,
-    versionFloat: parseFloat(`${match[1]}.${match[2]}`),
-    major: parseInt(match[1], 10),
-    minor: parseInt(match[2], 10),
-    patch: match[4] ? parseInt(match[4], 10) : undefined
+    versionString,
+    versionFloat: parseFloat(versionString),
```
Comment (mykola-mokhnach) [inline lib/xcode.js]: `parseFloat('1.2.3')` will not return what we expect

File: lib/xcode.js [hunk 2]
```diff
@@ -106,7 +107,7 @@ const getPath = _.memoize(function (timeout = XCRUN_TIMEOUT) {
 
 
 async function getVersionWithoutRetry (timeout = XCRUN_TIMEOUT) {
```
Comment (mykola-mokhnach) [inline lib/xcode.js:109]: is it an internal method?

File: lib/xcode.js [hunk 3]
```diff
@@ -136,20 +128,20 @@ const getVersionMemoized = _.memoize(
 );
 
 async function getVersion (parse = false, retries = DEFAULT_NUMBER_OF_RETRIES, timeout = XCRUN_TIMEOUT) {
-  let version = await getVersionMemoized(retries, timeout);
+  const version = await getVersionMemoized(retries, timeout);
+  // xcode version strings are not exactly semver string: patch versions of 0
+  // are removed (e.g., '10.0.0' => '10.0')
+  const versionString = version.patch > 0 ? version.version : `${version.major}.${version.minor}`;
```
Comment (mykola-mokhnach) [inline lib/xcode.js:134]: what is returned as `patch` if the last number is missing?

Comment (imurchie) [reply lib/xcode.js:134]:
```
 $ node
> const semver = require('semver')
undefined
> semver.coerce('10.0')
SemVer {
  loose: undefined,
  raw: '10.0.0',
  major: 10,
  minor: 0,
  patch: 0,
  prerelease: [],
  build: [],
  version: '10.0.0' }
```

File: lib/xcode.js (hunk 2 above)
Comment (imurchie) [reply lib/xcode.js:109]: Yes. It is only called within this module, and not exported.

File: lib/xcode.js (hunk 1 above)
Comment (mykola-mokhnach) [inline lib/xcode.js:141]: `parseFloat('10.0.1') = 10.0 = parseFloat('10.0.2')` - is it what we want to get?

Comment (imurchie) [reply lib/xcode.js:141]: Yes. This is how it's always been. Most of the code that uses Xcode version does not care about the patch version, and uses the float version.

**Pull request merged**